pkg/pgmodel/metadata/metadata.go

```diff
@@ -0,0 +1,215 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package metadata
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/timescale/promscale/pkg/pgmodel/common/schema"
+	"github.com/timescale/promscale/pkg/pgmodel/model"
+	"github.com/timescale/promscale/pkg/pgmodel/querier"
+	"github.com/timescale/promscale/pkg/pgxconn"
+)
+
+const (
+	metricMetadata = "SELECT metric_family, type, unit, help from " + schema.Catalog + ".metadata"
+	middle         = " ORDER BY last_seen"
+)
+
+func MetricMetadata(conn pgxconn.PgxConn, metric string, limit int) (map[string][]model.Metadata, error) {
+	query := metricMetadata
+	if metric != "" {
+		query += fmt.Sprintf(" WHERE metric_family='%s'", metric)
+	}
+	query += middle
+	if limit > 0 {
+		query += fmt.Sprintf(" LIMIT %d", limit)
```

Comment:
shouldn't this limit be on the number off metric families, not metadata entries?

Reply:
Yes, I missed this. But now, we need to fetch all metadata from db, then form the map. While forming the map, we respect the limit which is what Prometheus does.